qiskit/transpiler/passes/scheduling/asap.py

```diff
@@ -69,70 +48,99 @@ def run(self, dag):
         for creg in dag.cregs.values():
             new_dag.add_creg(creg)
 
-        qubit_time_available = defaultdict(int)
-        clbit_readable = defaultdict(int)
-        clbit_writeable = defaultdict(int)
-
-        def pad_with_delays(qubits: List[int], until, unit) -> None:
-            """Pad idle time-slots in ``qubits`` with delays in ``unit`` until ``until``."""
-            for q in qubits:
-                if qubit_time_available[q] < until:
-                    idle_duration = until - qubit_time_available[q]
-                    new_dag.apply_operation_back(Delay(idle_duration, unit), [q])
-
+        idle_after = {q: 0 for q in dag.qubits + dag.clbits}
         bit_indices = {q: index for index, q in enumerate(dag.qubits)}
         for node in dag.topological_op_nodes():
-            # validate node.op.duration
-            if node.op.duration is None:
-                indices = [bit_indices[qarg] for qarg in node.qargs]
-                if dag.has_calibration_for(node):
-                    node.op.duration = dag.calibrations[node.op.name][
-                        (tuple(indices), tuple(float(p) for p in node.op.params))
-                    ].duration
-
-                if node.op.duration is None:
-                    raise TranspilerError(
-                        f"Duration of {node.op.name} on qubits {indices} is not found."
-                    )
-            if isinstance(node.op.duration, ParameterExpression):
-                indices = [bit_indices[qarg] for qarg in node.qargs]
-                raise TranspilerError(
-                    f"Parameterized duration ({node.op.duration}) "
-                    f"of {node.op.name} on qubits {indices} is not bounded."
-                )
-            # choose appropriate clbit available time depending on op
-            clbit_time_available = (
-                clbit_writeable if isinstance(node.op, Measure) else clbit_readable
-            )
-            # correction to change clbit start time to qubit start time
-            delta = node.op.duration if isinstance(node.op, Measure) else 0
-            # must wait for op.condition_bits as well as node.cargs
-            start_time = max(
-                itertools.chain(
-                    (qubit_time_available[q] for q in node.qargs),
-                    (clbit_time_available[c] - delta for c in node.cargs + node.op.condition_bits),
-                )
-            )
-
-            pad_with_delays(node.qargs, until=start_time, unit=time_unit)
+            op_duration = self._get_node_duration(node, bit_indices, dag)
+
+            # compute t0, t1: instruction interval
+            if isinstance(node.op, Measure):
+                # measure instruction handling is bit tricky due to clbit_write_latency
+                t0q = max(idle_after[q] for q in node.qargs)
+                t0c = max(idle_after[c] for c in node.cargs)
+                if t0q > t0c:
+                    t0 = t0q
+                else:
```

Comment:
Similar to ASAP case, maybe `t0 = max(t0q, t0c - self.clbit_write_latency)` with your nice long comments.
qubits determine t0 => t0q,  clbits dterminde t0 => t0c - self.clbit_write_latency